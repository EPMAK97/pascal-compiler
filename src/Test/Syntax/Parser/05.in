var res : double;
procedure foo;
var x : integer;
Begin
    x := 2 + 2;
end;
Begin
    res := 1;
end.