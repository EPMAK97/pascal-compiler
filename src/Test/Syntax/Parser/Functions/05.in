program test;

procedure p(a: Integer);

function f : Integer;
begin
    Result := 10;
end;

begin
    a := a + f;
end;

function f(a, b, c: Integer) : Double;
begin
    Result := (a + b) / c;
end;

begin
end.