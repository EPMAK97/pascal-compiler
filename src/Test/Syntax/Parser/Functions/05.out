FUNCTION : test
    result : NIL
    function vars {
    FUNCTION : p
        result : NIL
        function params {
        a : INT
        }
        function vars {
        FUNCTION : f
            result : INT
            function params {
            }
            function vars {
            }
            └── statements
                         └── :=
                              ├── Result
                              └── 10

        }
        └── statements
                     └── :=
                          ├── a
                          └── +
                              ├── a
                              └── f

    FUNCTION : f
        result : DOUBLE
        function params {
        a : INT
        b : INT
        c : INT
        }
        function vars {
        }
        └── statements
                     └── :=
                          ├── Result
                          └── /
                              ├── +
                              |   ├── a
                              |   └── b
                              └── c

    }
    └── statements

